src/loaders/svg/tvgSvgSceneBuilder.cpp

```diff
@@ -356,6 +378,15 @@ static void _applyProperty(SvgNode* node, Shape* vg, float vx, float vy, float v
         vg->stroke(style->stroke.paint.color.r, style->stroke.paint.color.g, style->stroke.paint.color.b, style->stroke.opacity);
     }
 
+    if (node->style->clipPath.node && !node->style->clipPath.node->node.comp.userSpace) {
+        vg->bounds(&vx, &vy, &vw, &vh, false);
+        if (auto strokeW = vg->strokeWidth()) {
```

Comment:
```
<svg id="svg1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
<clipPath id="myClipOBB" clipPathUnits="objectBoundingBox" transform="matrix(1 0 0 1 -10 -10)">
<circle cx="0.5" cy="0.5" r="0.15"/>
</clipPath>

<clipPath id="myClipUS" transform="matrix(1 0 0 1 0 0)">
<circle cx="50" cy="50" r="15" />
</clipPath>


<rect x="25" y="25" width="50" height="50"  fill="red" clip-path="url(#myClipUS)"  fill-opacity="0.5" transform="matrix(1 0 0 1 0 0)" /> 

<rect x="0" y="0" width="100" height="100" fill="blue" clip-path="url(#myClipOBB)" fill-opacity="0.5" transform="matrix(1 0 0 1 10 10)" />
</svg>


```
Even if stroke is not declared, the default value is 1 .
This doesn't seem to be the intended result.

Reply:
@JSUYA thank you very much for the detailed rev.
please check my comments.

1. >Even if stroke is not declared, the default value is 1 .
This doesn't seem to be the intended result.

It's because the bounds() api is not the one that I really need here... but please lets keep it here as it is and I'll prepare a separate pr with some internal implementation of bounds for the svg loader. 
as you wrote, since the def value of the stroke width = 1, so this value is taken into account in the bounds() api, so I need to subtract this value here -> added in bounds, subtracted here. So the result is as it should be, but the way to get it should be improved :)


2. >The circles of the displayed result(chrome browser) have the same position and size.
But in our result, no blue circle is output.
As you said, w3school (or chrome) could be wrong.
But when I checked the final matrix value, there seems to be a problem with the translate value(dx, dy).

the result from tvg is as I'd expect. the `transform="matrix(1 0 0 1 -10 -10)"` is the same as `transform="translate(-10 -10)"`. When `objectBoundingBox` is used, these values are treated as they are percentages of the object's bbox. so for the rect from the example, we get: 100 * (-10) = -1000, and since you also transformed the path (+10), we got -1000 + 10 = -990. 

It works exactly the same for gradients (please notice the transformation matrix for the LinearGradient1):

Reply:
Ok
I understand that there is a problem with the transform of clippath in chrome and w3school.
I have one more question.

```
<svg id="svg1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
<clipPath id="myClipOBB" clipPathUnits="objectBoundingBox" >
<circle cx="0.5" cy="0.5" r="0.3" transform="matrix(1 0 0 1 0.1 0.1)"/>
</clipPath>

<clipPath id="myClipUS">
<circle cx="50" cy="50" r="30" transform="matrix(1 0 0 1 10 10)"/>
</clipPath>


<rect x="0" y="0" width="100" height="100"  fill="red" clip-path="url(#myClipUS)"  fill-opacity="0.5" transform="matrix(1 0 0 1 0 0)" /> 

<rect x="0" y="0" width="100" height="100" fill="blue" clip-path="url(#myClipOBB)" fill-opacity="0.5" transform="matrix(1 0 0 1 0 0)" />
</svg>

```

In this example, I think the transform applied to circle should be affected by objectBoundingBox.
In chrome, both circles are output the same. (This behavior is the same behavior we intend.)
but in thovg, the output is wrong.
( transform="matrix(1 0 0 1 0 0)"  output is also wrong)
It seems that the children of clippath(Scene) should be affected by OBB in the same way.
I know it's getting more and more complicated, too.

What do you think?

Reply:
and about the stroke..
If you use style->stroke.paint.none on line 349, you could be able to check whether or not there is a stroke.
What do you think?

Reply:
@JSUYA I'm so sorry that from the very beginning this patch is not perfect...
I was wrong about saying that clip shouldn't be a scene... It actually has to a scene, since otherwise we can not use different transformations for the shapes.

So I was wrong here:
https://github.com/Samsung/thorvg/pull/988#discussion_r742362843
clipPath is not a container element, but still it has to be a scene in our implementation.

I'll come back to this pr after #661 is merged

Reply:
> Ok I understand that there is a problem with the transform of clippath in chrome and w3school. I have one more question.
> 
> ```
> <svg id="svg1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
> <clipPath id="myClipOBB" clipPathUnits="objectBoundingBox" >
> <circle cx="0.5" cy="0.5" r="0.3" transform="matrix(1 0 0 1 0.1 0.1)"/>
> </clipPath>
> 
> <clipPath id="myClipUS">
> <circle cx="50" cy="50" r="30" transform="matrix(1 0 0 1 10 10)"/>
> </clipPath>
> 
> 
> <rect x="0" y="0" width="100" height="100"  fill="red" clip-path="url(#myClipUS)"  fill-opacity="0.5" transform="matrix(1 0 0 1 0 0)" /> 
> 
> <rect x="0" y="0" width="100" height="100" fill="blue" clip-path="url(#myClipOBB)" fill-opacity="0.5" transform="matrix(1 0 0 1 0 0)" />
> </svg>
> ```
> 
> In this example, I think the transform applied to circle should be affected by objectBoundingBox. In chrome, both circles are output the same. (This behavior is the same behavior we intend.) but in thovg, the output is wrong. ( transform="matrix(1 0 0 1 0 0)" output is also wrong) It seems that the children of clippath(Scene) should be affected by OBB in the same way. I know it's getting more and more complicated, too.
> 
> What do you think?

solved in #1532